sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -952,9 +942,25 @@ export class MessageReceiver extends LinkEntity {
           connectionHost: this._context.namespace.config.host,
           delayInSeconds: 15
         };
-        if (!this.wasCloseInitiated) {
-          await retry<void>(config);
-        }
+        await retry<void>(config);
+      } else if (causedByDisconnect) {
+        log.error(
+          "[%s] Something went wrong and the connection disconnected. Receiver '%s' with address '%s' encountered error: %O",
```

Comment:
When we know that the problem happened at the connection level, why are we saying that the receiver encountered the error?

```suggestion
          "[%s] Encountered a non retryable error on the connection. Cannot recover receiver '%s' with address '%s': %O",
```

Reply:
That's my poor word-smithing. In my head it was still the receiver that had an encounter with an error. Will update.